fdb-relational-jdbc/src/main/java/com/apple/foundationdb/relational/jdbc/JDBCRelationalConnection.java

```diff
@@ -345,7 +345,8 @@ public void setAutoCommit(boolean autoCommit) throws SQLException {
             this.autoCommit = false;
         } else {
             // commit any remaining work
-            commit();
+            // TODO do we need better protection if there is uncommitted work
+            //      does the autoCommit-ness need to be passed down
```

Comment:
This was doing the right thing
>   <B>NOTE:</B>  If this method is called during a transaction and the
> auto-commit mode is changed, the transaction is committed.  If
> <code>setAutoCommit</code> is called and the auto-commit mode is
> not changed, the call is a no-op.

But, it would end up calling `commit` on the server when there was no active transaction, which failed.